engine/src/test/java/com/datatorrent/stram/engine/GenericNodeTest.java

```diff
@@ -393,6 +401,119 @@ public void run()
+
+    final String streamName = "streamName";
+    final String upstreamNodeId = "upstreamNodeId";
+    final String  downstreamNodeId = "downStreamNodeId";
+
+    StreamContext issContext;
+    StreamContext ossContext;
+    BufferServerSubscriber iss;
+    BufferServerPublisher oss;
+
+    EventLoop eventloop = DefaultEventLoop.createEventLoop("StreamTestEventLoop");
+
+    ((DefaultEventLoop)eventloop).start();
+    bufferServer = new Server(0); // find random port
+    InetSocketAddress bindAddr = bufferServer.run(eventloop);
+    bufferServerPort = bindAddr.getPort();
+
+    final StreamCodec<Object> serde = new DefaultStatefulStreamCodec<Object>();
+    final ArrayList<Object> list = new ArrayList<Object>();
+
+    GenericOperator go = new GenericOperator();
+    final GenericNode gn = new GenericNode(go, new com.datatorrent.stram.engine.OperatorContext(0, "operator",
+        new DefaultAttributeMap(), null));
+    gn.setId(1);
+
+    Sink<Object> output = new Sink<Object>()
+    {
+      @Override
+      public void put(Object tuple)
+      {
+        list.add(tuple);
+      }
+
+      @Override
+      public int getCount(boolean reset)
+      {
+        return 0;
+      }
+    };
+
+    issContext = new StreamContext(streamName);
+    issContext.setSourceId(upstreamNodeId);
+    issContext.setSinkId(downstreamNodeId);
+    issContext.setFinishedWindowId(-1);
+    issContext.setBufferServerAddress(InetSocketAddress.createUnresolved("localhost", bufferServerPort));
+    issContext.put(StreamContext.CODEC, serde);
+    issContext.put(StreamContext.EVENT_LOOP, eventloop);
+
+    iss = new BufferServerSubscriber(downstreamNodeId, 1024);
+    iss.setup(issContext);
+    iss.activate(issContext);
+
+    ossContext = new StreamContext(streamName);
+    ossContext.setSourceId(upstreamNodeId);
+    ossContext.setSinkId(downstreamNodeId);
+    ossContext.setBufferServerAddress(InetSocketAddress.createUnresolved("localhost", bufferServerPort));
+    ossContext.put(StreamContext.CODEC, serde);
+    ossContext.put(StreamContext.EVENT_LOOP, eventloop);
+
+    oss = new BufferServerPublisher(upstreamNodeId, 1024);
+    oss.setup(ossContext);
+    oss.activate(ossContext);
+
+    gn.connectInputPort("ip1", iss.acquireReservoir("testReservoir", 1));
+    gn.connectOutputPort("op", output);
+    gn.firstWindowMillis = 0;
+    gn.windowWidthMillis = 100;
+
+    Tuple beginWindow1 = new Tuple(MessageType.BEGIN_WINDOW, 0x1L);
+    Tuple endWindow1 = new EndWindowTuple(0x1L);
+    Tuple beginWindow2 = new Tuple(MessageType.BEGIN_WINDOW, 0x2L);
+    Tuple endWindow2 = new EndWindowTuple(0x2L);
+    Tuple beginWindow3 = new Tuple(MessageType.BEGIN_WINDOW, 0x3L);
+    Tuple endWindow3 = new EndWindowTuple(0x3L);
+    EndStreamTuple est = new EndStreamTuple(0L);
+
+    oss.put(beginWindow1);
+    oss.put(endWindow1);
+    oss.put(beginWindow2);
+    oss.put(endWindow2);
+    oss.put(beginWindow3);
+    oss.put(endWindow3);
+    oss.put(est);
```

Comment:
The purpose of the test is to check that BufferServerSubscriber can be activated and process tuples without a loss even if it is activated before the operator is setup and starts processing tuples. There is no guarantee that Publisher actually delivers tuples to the buffer server and buffer server sends them to the Subscriber in the test.